tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/KafkaConsumerConstructorIntegration.cs

```diff
@@ -51,11 +53,25 @@ public class KafkaConsumerConstructorIntegration
                 }
             }
 
+            if (instance != null)
+            {
+                // we need to get the actual handler type to make this work in case
+                // the original handler is not set
+                var handlerType = ConsumerCache.GetOffsetsCommittedHandlerType(instance);
+                if (handlerType != null)
+                {
+                    consumer.OffsetsCommittedHandler = DelegateInstrumentation.Wrap(
+                        consumer.OffsetsCommittedHandler,
+                        handlerType,
+                        new OffsetsCommittedCallbacks(groupId));
+                }
```

Comment:
What's the original type of `consumer.OffsetsCommittedHandler` ? if the original type is the actual Handler type, you can use the `ValueWithType<Delegate>` trick in the ducktype proxy.

There's an example here: 
https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/test/Datadog.Trace.Tests/Util/Delegates/ActionInstrumentationTests.cs#L458-L491

More info here: https://github.com/DataDog/dd-trace-dotnet/pull/4621

You can instrument a `ValueWithType<Delegate>` delegate directly using the extension method: https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/Util/Delegates/DelegatesExtensions.cs#L28-L39

Reply:
Thanks for the suggestion, I'll update the PR.